internal/handlers/telegram/handler.go

```diff
@@ -67,12 +68,45 @@ func messageHandler(tg *Client, u tgbotapi.Update) {
 		return
 	}
 
-	formatted := fmt.Sprintf("%s%s%s %s",
+	// Telegram user replied to a message
+	if u.Message.ReplyToMessage != nil {
+		replyHandler(tg, u)
+		return
+	}
+
+	formatted = fmt.Sprintf("%s%s%s %s",
 		tg.Settings.Prefix,
 		username,
 		tg.Settings.Suffix,
 		// Trim unexpected trailing whitespace
 		strings.Trim(u.Message.Text, " "))
+
+	tg.sendToIrc(formatted)
+}
+
+/*
+replyHandler handles when users reply to a Telegram message
+*/
+func replyHandler(tg *Client, u tgbotapi.Update) {
+	replyText := strings.Trim(u.Message.ReplyToMessage.Text, " ")
+	username := GetUsername(tg.IRCSettings.ShowZWSP, u.Message.From)
+	replyUser := GetUsername(tg.IRCSettings.ShowZWSP, u.Message.ReplyToMessage.From)
+
+	// Only show a portion of the reply text
+	if len(replyText) > tg.Settings.ReplyLength {
+		replyText = replyText[0:tg.Settings.ReplyLength] + "…"
+	}
+
+	formatted := fmt.Sprintf("%s%s%s %sRe %s: %s%s %s",
```

Comment:
It might be better to make the `Re` part of the reply prefix

Reply:
Hmm, do you think this is a big enough of a QoL that it would be a blocker to this PR not to add it in?

Reply:
Nah, I don't think it's a big deal